### PR TITLE
#971 : Media support with BuddyPress message.

### PR DESCRIPTION
**Pull request by malavvasita**
We do not provide Media attachments to BuddyPress personal messages right now. We can add this functionality in rtMedia.

Issue: https://github.com/rtMediaWP/rtMedia/issues/971

### CODE REVIEW

File: app/assets/css/sass/rtmedia.scss [hunk 1]
```diff
@@ -39,3 +39,6 @@
 
 // *** RTL *** //
 @import "rtl";
+
+// *** BP Media Message *** //
+@import "media-message";
```
Comment (thrijith) [inline app/assets/css/sass/rtmedia.scss]: i don't see any **scss** file for this `@import "media-message";`
if not used please remove.

File: app/assets/js/rtMedia.backbone.js [hunk 2]
```diff
@@ -497,6 +498,13 @@ jQuery( function( $ ) {
 			},
 			initialize: function( config ) {
 				this.uploader = new plupload.Uploader( config );
+                                /*
+					var current_url will fetch present working area's address and we will find if it contains message in url though find valirable. If message is there in URL then we need to mention that it is message. So we have appended message:true in config.
```
Comment (thrijith) [inline app/assets/js/rtMedia.backbone.js]: break comment into multiple lines

also follow comment structure as discussed.

File: app/assets/js/rtMedia.backbone.js [hunk 3]
```diff
@@ -497,6 +498,13 @@ jQuery( function( $ ) {
...
+				*/
+				var current_url = document.URL, find= 'message';
+				if(current_url.indexOf(find) !== -1){
```
Comment (thrijith) [inline app/assets/js/rtMedia.backbone.js]: follow WordPress Coding Standards

File: app/assets/js/rtMedia.backbone.js [hunk 4]
```diff
@@ -556,6 +564,12 @@ jQuery( function( $ ) {
 			uploaderObj.initUploader();
 
 			uploaderObj.uploader.bind( 'UploadComplete', function( up, files ) {
+				// Success message for BudyyPress Media Message
+				if ( jQuery( '.rtm-media-msg-upload-button' ).length == 1 ) {
+					jQuery( '.rtm-media-msg-upload-button' ).html( "" );
+					jQuery( '.rtm-media-msg-upload-button' ).removeAttr( "id" );
+					jQuery( '.rtm-media-msg-upload-button' ).html( "<p id='rtm_bpm_success' style='background: #98ef98; padding: 20px;'>Media has been attached with this message!</p>" );
```
Comment (thrijith) [inline app/assets/js/rtMedia.backbone.js]: create HTML elements using jQuery.

File: app/assets/js/rtMedia.backbone.js [hunk 5]
```diff
@@ -795,6 +809,11 @@ jQuery( function( $ ) {
 					return message;
 				};
 			} );
+				
+			// This function will store hidden values of Media ID in Array
+        	function store_array_in_hidden_field(){
+				jQuery("#rtm_bpm_uploaded_media").attr("value", msg_media_files.toString());
```
Comment (thrijith) [inline app/assets/js/rtMedia.backbone.js]: `jQuery("#rtm_bpm_uploaded_media").attr("value", msg_media_files.toString());` could be 

`jQuery( '#rtm_bpm_uploaded_media' ).attr( 'value', msg_media_files.toString() );`

File: app/assets/js/rtMedia.backbone.js [hunk 6]
```diff
@@ -826,6 +845,16 @@ jQuery( function( $ ) {
 			} );
 
 			uploaderObj.uploader.bind( 'FileUploaded', function( up, file, res ) {
+                                var uploaded_response_data = JSON.parse(res.response);
```
Comment (thrijith) [inline app/assets/js/rtMedia.backbone.js]: align code and add spaces

File: app/assets/js/rtMedia.backbone.js [hunk 7]
```diff
@@ -826,6 +845,16 @@ jQuery( function( $ ) {
...
+				if(uploaded_response_data.length<=0){
```
Comment (thrijith) [inline app/assets/js/rtMedia.backbone.js]: Add spaces

File: app/assets/js/rtMedia.backbone.js [hunk 8]
```diff
@@ -826,6 +845,16 @@ jQuery( function( $ ) {
...
+					jQuery( '.rtm-media-msg-upload-button' ).html("");
+					jQuery( '.rtm-media-msg-upload-button' ).removeAttr( "id" );
+					jQuery( '.rtm-media-msg-upload-button' ).html("<p style='background: #db001e; padding: 20px; color:white;'>Media attachement failed! Please try again!</p>");
```
Comment (thrijith) [inline app/assets/js/rtMedia.backbone.js]: create HTML elements using jQuery

File: app/main/controllers/shortcodes/RTMediaGalleryShortcode.php [hunk 9]
```diff
@@ -68,6 +68,16 @@ static function register_scripts() {
...
 		$url         = rtmedia_get_upload_url( $request_uri );
 
+                /*
+                    @param array $url_array
+                    @param mixed $url
+                    Redirects to 'upload.php' when we are composing message and uplaod media with BP message
+                */
+                if (strpos($url, '/messages') !== false) {
+                    $url_array = explode('/messages', $url);
+                    $url = trailingslashit($url_array[0]).'upload/';
```
Comment (thrijith) [inline app/main/controllers/shortcodes/RTMediaGalleryShortcode.php]: add spaces

File: templates/media/media-with-message.php [hunk 10]
```diff
@@ -0,0 +1,151 @@
...
+ * @package rtMedia
+ **/
+
+/**
+ *  This function will place uploader in BuddyPress compose message form and handle send reply activity.
+ */
+function rtm_bp_message_media_add_upload_media_button() {
+?>
+	<script>
+		$(function(){
```
Comment (thrijith) [inline templates/media/media-with-message.php]: either use `$` or `jQuery` maintain consistency

File: templates/media/media-with-message.php [hunk 11]
```diff
@@ -0,0 +1,151 @@
...
+			$("#send-notice").click(function () {
+				if ($(this).is(":checked")) {
+					$("#rtm_show_upload_ui").slideUp();
+				} else {
+					$("#rtm_show_upload_ui").slideDown();
+				}
+			});
+		});
+		jQuery(document).ready(function(){
```
Comment (thrijith) [inline templates/media/media-with-message.php]: add spaces

File: templates/media/media-with-message.php [hunk 12]
```diff
@@ -0,0 +1,151 @@
...
+			handler = function(){
+				var order = jQuery('#messages_order').val() || 'ASC',
+				offset = jQuery('#message-recipients').offset();
```
Comment (thrijith) [inline templates/media/media-with-message.php]: align `=`

File: templates/media/media-with-message.php [hunk 13]
```diff
@@ -0,0 +1,151 @@
...
+					}
+					jQuery(button).removeClass('loading');
+				});
+				return false;
+			};
+			$( "input#send_reply_button" ).unbind( "click").bind("click", handler);
+		});
+
+	</script>
+	<span class="primary rtm-media-msg-upload-button rtmedia-upload-media-link" id="rtm_show_upload_ui" title="Upload Media"><i class="dashicons dashicons-upload rtmicon"></i>Upload Media File</span>
```
Comment (thrijith) [inline templates/media/media-with-message.php]: break into more lines for better readability.

File: templates/media/media-with-message.php [hunk 14]
```diff
@@ -0,0 +1,151 @@
...
+	</div>
+	<input type="hidden" id="rtm_bpm_uploaded_media" name="rtm_bpm_uploaded_media" />
+	<?php
+}
+
+/**
+	Add additional parameter media ID in BuddyPress message SEND MESSAGE process. and Insert data into MEDIA_META table.
+
+	@param object $message Getting parameter response when sending message.
+ **/
```
Comment (thrijith) [inline templates/media/media-with-message.php]: comment structure

File: templates/media/media-with-message.php [hunk 15]
```diff
@@ -0,0 +1,151 @@
...
+				]
+			);
+		}
+	}
+	?>
+	<script>
+		jQuery("#msg-success-bp-msg-media").hide();
+		jQuery(".rtm-media-msg-upload-button").attr("id", "rtm_show_upload_ui");
+		jQuery(".rtm-media-msg-upload-button").html("");
+		jQuery(".rtm-media-msg-upload-button").html("<i class='dashicons dashicons-upload rtmicon'></i>Upload Media File");
```
Comment (thrijith) [inline templates/media/media-with-message.php]: create HTML elements using jQuery

File: templates/media/media-with-message.php [hunk 16]
```diff
@@ -0,0 +1,151 @@
...
+	</script>
+	<?php
+}
+
+/**
+
```
Comment (thrijith) [inline templates/media/media-with-message.php]: comment structure

File: templates/media/media-with-message.php [hunk 17]
```diff
@@ -0,0 +1,151 @@
...
+}
+
+// Adding rtMedia uploader in BuddyPress compose message form.
+add_action( 'bp_after_messages_compose_content', 'rtm_bp_message_media_add_upload_media_button' );
+// Adding rtMedia uploader in BuddyPress Send Reply form.
+add_action( 'bp_after_message_reply_box', 'rtm_bp_message_media_add_upload_media_button' );
+// Handling BuddyPress send message action by adding MEDIA ID.
+add_action( 'messages_message_sent', 'rtm_add_message_media_params' );
+// Showing media below BuddyPress message.
+add_action( 'bp_after_message_content', 'show_rtm_bp_msg_media' );
```
Comment (thrijith) [inline templates/media/media-with-message.php]: add new line at the end of file

File: templates/media/media-with-message.php [hunk 18]
```diff
@@ -0,0 +1,155 @@
...
+	<span class="primary rtm-media-msg-upload-button rtmedia-upload-media-link" 
+		id="rtm_show_upload_ui" title="Upload Media">
+		<i class="dashicons dashicons-upload rtmicon"></i>
+			Upload Media File
+	</span>
+	<div id="rtm-media-gallery-uploader" class="rtm-media-gallery-uploader">
+		<?php
+		rtmedia_uploader(
+			[
+				'is_up_shortcode' => false,
```
Comment (thrijith) [inline templates/media/media-with-message.php]: remove extra spaces

File: templates/media/media-with-message.php [hunk 19]
```diff
@@ -0,0 +1,155 @@
...
+				]
+			);
+		}
+	}
+	?>
+	<script>
+		jQuery( '#msg-success-bp-msg-media' ).hide();
+		jQuery( '.rtm-media-msg-upload-button' ).attr( 'id', 'rtm_show_upload_ui' );
+		jQuery( '.rtm-media-msg-upload-button' ).html( '' );
+		jQuery( '.rtm-media-msg-upload-button' ).html( jQuery( '<i>', { class: 'dashicons dashicons-upload rtmicon' }{ text: 'Upload Media File' } ) );
```
Comment (thrijith) [inline templates/media/media-with-message.php]: break into more lines for readability

File: templates/media/media-with-message.php [hunk 20]
```diff
@@ -0,0 +1,155 @@
...
+	if ( '0' !== $media_result[0]->media_id ) {
+	?>
+	<ul class='rtmedia-list-media rtm-gallery-list clearfix' style = 'margin-top: 10px;'>
+	<?php
+	foreach ( $media_result as $media_result_array_value ) {
+		$media     = rtmedia_image( 'rt_media_thumbnail', $media_result_array_value->media_id, false );
+		$media_url = $url[0] . '/media/' . $media_result_array_value->media_id . '/';
+		?>
+
+			<li class='rtmedia-list-item' style='display:inline; float: left;'' id="<?php echo $media_result_array_value->media_id; // @codingStandardsIgnoreLine?>">
```
Comment (thrijith) [inline templates/media/media-with-message.php]: don't use inline css if possible 
use `esc_attr` for id
`style='display:inline; float: left;''` ?

File: templates/media/media-with-message.php [hunk 21]
```diff
@@ -0,0 +1,161 @@
...
+						jQuery( 'form#send-reply' ).prepend( response.substr( 2, response.length ) );
+					} else {
+						jQuery( 'form#send-reply div#message' ).remove();
+						jQuery( '#message_content' ).val( '' );
+						jQuery( '#rtm_bpm_uploaded_media' ).removeAttr( 'value' );
+						if ( 'ASC' == order ) {
+							jQuery( 'form#send-reply' ).before( response );
+						} else {
+							jQuery( '#message-recipients' ).after( response );
+							jQuery(window).scrollTop(offset.top);
```
Comment (thrijith) [inline templates/media/media-with-message.php]: `jQuery( window ).scrollTop( offset.top );`

File: templates/media/media-with-message.php [hunk 22]
```diff
@@ -0,0 +1,161 @@
...
+function rtm_add_message_media_params( $message ) {
+	$insert_media_object  = new RTDBModel( 'rtm_media_meta' );
+	$message->media_array = filter_input( INPUT_POST, 'rtm_bpm_uploaded_media' );
+	$media                = explode( ',', $message->media_array );
+	if ( ! empty( $media ) && null !== $media ) {
+		foreach ( $media as $media_id ) {
+			$insert_media_object->insert(
+				[
+					'media_id'   => $media_id,
+					// Adding meta data into custom meta table
```
Comment (thrijith) [inline templates/media/media-with-message.php]: add `.` at the end of comments, sorry for not pointing out first

File: app/assets/js/rtMedia.backbone.js [hunk 23]
```diff
@@ -497,7 +497,14 @@ jQuery( function( $ ) {
 			},
 			initialize: function( config ) {
 				this.uploader = new plupload.Uploader( config );
-				/*
+				
+				//current_url will fetch present URL
+				var current_url = document.URL, find = 'message';
+				if( -1 !== current_url.indexOf( find ) ){
```
Comment (chandrapatel) [inline app/assets/js/rtMedia.backbone.js]: Don't create un-necessary variables. Here `find` var not used anywhere else.

Comment (chandrapatel) [reply app/assets/js/rtMedia.backbone.js]: Please follow WordPress JS coding standards. Please follow it in your changes as of now.

File: app/assets/js/rtMedia.backbone.js [hunk 24]
```diff
@@ -556,6 +563,16 @@ jQuery( function( $ ) {
 			uploaderObj.initUploader();
 
 			uploaderObj.uploader.bind( 'UploadComplete', function( up, files ) {
+				// Success message for BudyyPress Media Message
+				if ( jQuery( '.rtm-media-msg-upload-button' ).length == 1 ) {
+                                    jQuery( '.rtm-media-msg-upload-button' ).html( '' );
+                                    jQuery( '.rtm-media-msg-upload-button' ).removeAttr( 'id' );
+                                    jQuery( '.rtm-media-msg-upload-button' ).html(
```
Comment (chandrapatel) [inline app/assets/js/rtMedia.backbone.js]: Use `.append()` method

File: app/assets/js/rtMedia.backbone.js [hunk 25]
```diff
@@ -825,7 +842,28 @@ jQuery( function( $ ) {
 					rtMediaHook.call( 'rtmedia_js_before_file_upload', [up, file] );
 			} );
 
-			uploaderObj.uploader.bind( 'FileUploaded', function( up, file, res ) {
+                                uploaderObj.uploader.bind( 'FileUploaded', function( up, file, res ) {
+                                    var uploaded_response_data = JSON.parse( res.response );
+                                    if ( uploaded_response_data.length <= 0 ) {
+					jQuery( '.rtm-media-msg-upload-button' ).html( '' );
+					jQuery( '.rtm-media-msg-upload-button' ).removeAttr( 'id' );
+					jQuery( '.rtm-media-msg-upload-button' ).html(
```
Comment (chandrapatel) [inline app/assets/js/rtMedia.backbone.js]: Use `.append()` method.

File: app/main/RTMedia.php [hunk 26]
```diff
@@ -1089,6 +1089,8 @@ function enqueue_scripts_styles() {
 		// Localizing strings for rtMedia.backbone.js
 		$rtmedia_backbone_strings = array(
 			'rtm_edit_file_name' => esc_html__( 'Edit File Name', 'buddypress-media' ),
+			'rtm_bp_msg_media_success' => esc_html__( 'Media has been attached with this message!', 'buddypress-media' ),
+			'rtm_bp_msg_media_failure' => esc_html__( 'Media attachement failed! Please try again!', 'buddypress-media' )
```
Comment (chandrapatel) [inline app/main/RTMedia.php]: Add `,` at the end of array item. PHPCS gave warning.

File: templates/media/media-with-message.php [hunk 27]
```diff
@@ -0,0 +1,161 @@
+<?php
+/**
+ * Functionalities provided in this page will allow users to send media with BuddyPress message.
+ *
+ * @author malav vasita malav.vasita@rtcamp.com
```
Comment (chandrapatel) [inline templates/media/media-with-message.php]: `@author Malav Vasita <malav.vasita@rtcamp.com>`

File: templates/media/media-with-message.php [hunk 28]
```diff
@@ -0,0 +1,161 @@
...
+ * @package rtMedia
+ **/
+
+/**
+ *  This function will place uploader in BuddyPress compose message form and handle send reply activity.
+ */
+function rtm_bp_message_media_add_upload_media_button() {
+?>
+	<script>
+		jQuery( function(){
```
Comment (chandrapatel) [inline templates/media/media-with-message.php]: Please follow WordPress JS Coding Standard. Please check your js code.

File: templates/media/media-with-message.php [hunk 29]
```diff
@@ -0,0 +1,161 @@
...
+
+/**
+ * Add additional parameter media ID in BuddyPress message SEND MESSAGE process.
+ * And Insert data into MEDIA_META table.
+ *
+ * @param object $message Getting parameter response when sending message.
+ **/
+function rtm_add_message_media_params( $message ) {
+	$insert_media_object  = new RTDBModel( 'rtm_media_meta' );
+	$message->media_array = filter_input( INPUT_POST, 'rtm_bpm_uploaded_media' );
```
Comment (chandrapatel) [inline templates/media/media-with-message.php]: You also need to pass `Sanitize filters` based on data type as third argument. or you can use `sanitize_text_field ()`

File: templates/media/media-with-message.php [hunk 30]
```diff
@@ -0,0 +1,161 @@
...
+	if ( '0' !== $media_result[0]->media_id ) {
+	?>
+	<ul class='rtmedia-list-media rtm-gallery-list clearfix' style = 'margin-top: 10px;'>
+	<?php
+	foreach ( $media_result as $media_result_array_value ) {
+		$media     = rtmedia_image( 'rt_media_thumbnail', $media_result_array_value->media_id, false );
+		$media_url = $url[0] . '/media/' . $media_result_array_value->media_id . '/';
+		?>
+
+			<li class='rtmedia-list-item rtmedia_bp_msg_media_upload_render' id="<?php echo $media_result_array_value->media_id; // @codingStandardsIgnoreLine?>">
```
Comment (chandrapatel) [inline templates/media/media-with-message.php]: You need to escape output using `esc_attr()` and remove `// @codingStandardsIgnoreLine`

File: templates/media/media-with-message.php [hunk 31]
```diff
@@ -0,0 +1,161 @@
...
+				<a href="<?php echo esc_url( $media_url ); ?>" class="<?php echo esc_attr( apply_filters( 'rtmedia_gallery_list_item_a_class', 'rtmedia-list-item-a' ) ); ?>">
```
Comment (chandrapatel) [inline templates/media/media-with-message.php]: Don't use `apply_filters()` within some function. It should be in new line with proper documentation. Save the result in a variable and then use it.

File: templates/media/media-with-message.php [hunk 32]
```diff
@@ -0,0 +1,161 @@
...
+					<div class="rtmedia-item-thumbnail">
+						<img src="<?php echo esc_url( $media ); ?>" alt="<?php echo esc_attr( apply_filters( 'rtmc_change_alt_text', $alt_text, $rtmedia_media ) ); ?>">
```
Comment (chandrapatel) [inline templates/media/media-with-message.php]: Don't use `apply_filters()` within some function. It should be in new line with proper documentation. Save the result in a variable and then use it.

File: templates/media/media-with-message.php (hunk 17 above)
Comment (chandrapatel) [reply templates/media/media-with-message.php]: Add new line after each `add_action()` call.

File: app/assets/js/rtMedia.backbone.js [hunk 33]
```diff
@@ -497,7 +497,14 @@ jQuery( function( $ ) {
 			},
 			initialize: function( config ) {
 				this.uploader = new plupload.Uploader( config );
-				/*
+				
+				//current_url will fetch present URL
+				var current_url = document.URL;
+				if( -1 !== current_url.indexOf( 'message' ) ){
```
Comment (chandrapatel) [inline app/assets/js/rtMedia.backbone.js]: Change it to `if ( -1 !== current_url.indexOf( 'message' ) ) {`

Please follow WordPress JS coding standards to keep code consistent.

File: app/assets/js/rtMedia.backbone.js [hunk 34]
```diff
@@ -825,7 +842,28 @@ jQuery( function( $ ) {
...
+					jQuery( '.rtm-media-msg-upload-button' ).append( 
+						jQuery( '<p>', { id: 'rtm_bpm_success' } )
+					);
+					jQuery( '#rtm_bpm_success' ).css(
+						{ 'background: #db001e':'#db001e', 'padding':'20px' }
+					);
+					jQuery( '#rtm_bpm_success' ).append( 
+						rtmedia_backbone_strings.rtm_bp_msg_media_failure 
+					);
+				}else{
```
Comment (chandrapatel) [inline app/assets/js/rtMedia.backbone.js]: `} else {`

File: templates/media/media-with-message.php [hunk 35]
```diff
@@ -0,0 +1,173 @@
+<?php
+/**
+ * Functionalities provided in this page will allow users to send media with BuddyPress message.
+ *
+ * @author Malav Vasita malav.vasita@rtcamp.com
```
Comment (chandrapatel) [inline templates/media/media-with-message.php]: `@author Malav Vasita <malav.vasita@rtcamp.com>`

File: templates/media/media-with-message.php [hunk 36]
```diff
@@ -0,0 +1,173 @@
...
+ * @param object $message Getting parameter response when sending message.
+ **/
+function rtm_add_message_media_params( $message ) {
+	$insert_media_object  = new RTDBModel( 'rtm_media_meta' );
+	$message->media_array = sanitize_text_field( filter_input( INPUT_POST, 'rtm_bpm_uploaded_media' ) );
+	$media                = explode( ',', $message->media_array );
+	if ( ! empty( $media ) && null !== $media ) {
+		foreach ( $media as $media_id ) {
+			$insert_media_object->insert(
+				[
```
Comment (chandrapatel) [inline templates/media/media-with-message.php]: Don't use this `[]` array format. It won't work with PHP version <= 5.3. Use `array()` syntax instead.

File: templates/media/media-with-message.php [hunk 37]
```diff
@@ -0,0 +1,173 @@
...
+				]
+			);
+		}
+	}
+	?>
+	<script>
+		jQuery( '#msg-success-bp-msg-media' ).hide();
+		jQuery( '.rtm-media-msg-upload-button' ).attr( 'id', 'rtm_show_upload_ui' );
+		jQuery( '.rtm-media-msg-upload-button' ).html( '' );
+		jQuery( '.rtm-media-msg-upload-button' ).html(
```
Comment (chandrapatel) [inline templates/media/media-with-message.php]: Here also, you need to use `append()` method.

File: templates/media/media-with-message.php [hunk 38]
```diff
@@ -0,0 +1,173 @@
...
+ * @var array  $url                        Partitions of current URL to redirect to perticular media.
+ * @var mixed  $rtm_gallary_list_filter    Filter for Media gallary listing.
+ * @var mixed  $rtm_change_alt_text_filter Filter for changing alter text of an image.
+ **/
+function show_rtm_bp_msg_media() {
+	$get_data_object            = new RTDBModel( 'rtm_media_meta' );
+	$media_result               = $get_data_object->get( [ 'meta_value' => bp_get_the_thread_message_id() ] );  // phpcs:ignore
+	$url                        = explode( 'messages/', sanitize_text_field( wp_unslash( filter_input( INPUT_SERVER, 'REQUEST_URI' ) ) ) );
+	$rtm_gallary_list_filter    = apply_filters( 'rtmedia_gallery_list_item_a_class', 'rtmedia-list-item-a' );
+	$rtm_change_alt_text_filter = apply_filters( 'rtmc_change_alt_text', $alt_text, $rtmedia_media );
```
Comment (chandrapatel) [inline templates/media/media-with-message.php]: I told you to add inline documentation for these new filters.

File: templates/media/media-with-message.php [hunk 39]
```diff
@@ -0,0 +1,173 @@
...
+	</script>
+	<span class="primary rtm-media-msg-upload-button rtmedia-upload-media-link" 
+		id="rtm_show_upload_ui" title="Upload Media">
+		<i class="dashicons dashicons-upload rtmicon"></i>
+			<?php echo esc_html__( 'Upload Media File', 'buddypress-media' ); ?>
+	</span>
+	<div id="rtm-media-gallery-uploader" class="rtm-media-gallery-uploader">
+		<?php
+		rtmedia_uploader(
+			[
```
Comment (chandrapatel) [inline templates/media/media-with-message.php]: Don't use this `[]` array syntax. It won't work with PHP version <= 5.3. Use `array()` syntax instead.

File: templates/media/media-with-message.php (hunk 38 above)
Comment (malavasita) [reply templates/media/media-with-message.php]: Filters are already there in rtmedia-filters... I have just re-used them. 
So I have mention why I have used them in Function Doc Comment above.

/**
 *As a result show attached media with message.
 *
 *@var object $get_data_object            Object for having DB actions.
 *@var array  $media_result               Getting media_ids attached with messages.
 *@var array  $url                        Partitions of current URL to redirect to perticular media.
 *@var mixed  $rtm_gallary_list_filter    Filter for Media gallary listing.
 *@var mixed  $rtm_change_alt_text_filter Filter for changing alter text of an image.
 **/

**Comment by yahilmadakiya** [comment]
Found some issues in this PR.

1. Can not send only Media, User have to add message as well.
2. Currently, it requires 2 button events to send message with media. We can improve this by adding a single button instead.
3. Message media should not visible at user profile `members/admin/media/`.

**Pull request closed without merging**